lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -523,6 +523,36 @@ def getvm(self, name=None, uuid=None, folder=None):
 
         return vm
 
+    def get_datacenter(self):
+        # datacenters = get_all_objs(self.content, [vim.Datacenter])
+        datacenter = self.cache.find_obj(self.content, [vim.Datacenter], self.params['datacenter'])
+        if datacenter is None:
+            self.module.fail_json(msg='No datacenter named %(datacenter)s was found' % self.params)
+
+        return datacenter
+
+    def get_folder_path(self, datacenter):
+        # Prepend / if it was missing from the folder path, also strip trailing slashes
+        if not self.params['folder'].startswith('/'):
```

Comment:
I would do:
```python
    # Ensure we have a starting slash and no trailing slash
    self.params['folder'] = os.path.join('/', self.params['folder']).rstrip('/')
```

Reply:
Is `posixpath.join()` better than `os.path.join()`?

Reply:
Yes :-)